### ISSUE TITLE
Tool Call Repair Doesn't Support `streamText`

### ISSUE THREAD

**Issue by mutewinter** [vercel/ai#4111]
### Description

The new `experimental_repairToolCall` function introduced in #3865 does not seem to support `streamText`. This appears to be due to an early exit inside `runToolsTransformation`:

https://github.com/vercel/ai/blob/bccd428efb312f06059f14c002023ba8f3c1c5f8/packages/ai/core/generate-text/run-tools-transformation.ts#L183-L202

If the tool isn't found, this code causes the transformation to exit early and not call the subsequent `parseToolCall`, which contains the repair logic.

### Code example

   ```ts
    streamText({
      experimental_repairToolCall: async (params) => {
        // Never called
      },
      messages,
      model,
      system: [
        "You are a tool calling assistant.",
      ].join("\n"),
      tools: {
        weather: tool({
          description: "Provides the weather for a given location",
          parameters: z.object({
            location: z.string(),
          }),
        }),
      },
    });
```

### AI provider

Any

**Comment by lgrammel** [comment]
Fixed in `ai@4.0.20`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This describes a concrete failure of an existing API capability with `streamText`, subsequently confirmed and fixed by a maintainer.

### Evidence
- `experimental_repairToolCall` was not invoked when used with `streamText` because tool transformation exited before reaching the repair logic.
- The reporter supplied a concrete reproduction and identified the failing control flow.
- A maintainer confirmed the issue was fixed in `ai@4.0.20`.
